packages/ui/src/components/UnitInput/index.tsx

```diff
@@ -0,0 +1,329 @@
+    }
+    & > ${StyledInputWrapper}:hover {
+      border-right-color: ${({ theme }) => theme.colors.danger.border};
+    }
+  }
+  &[data-error='true'] {
+    border: 1px solid ${({ theme }) => theme.colors.danger.border};
+    & > ${StyledInputWrapper} {
+      border-right-color: ${({ theme }) => theme.colors.danger.border};
+    }
+  }
+`
+
+type UnitInputValue = { inputValue: number; unit: string }
+
+type UnitInputProps = {
+  id?: string
+  className?: string
+  name: string
+  disabled?: boolean
+  maxValue?: number
+  minValue?: number
+  value?: UnitInputValue['inputValue']
+  unitValue?: UnitInputValue['unit']
+  onChange: (value: UnitInputValue['inputValue']) => void
+  onChangeUnitValue: (values: string[]) => void
+  options: OptionType[]
+  placeholder?: string
+  selectInputWidth?: number
+  size?: 'small' | 'medium' | 'large'
+  'data-testid'?: string
+  notice?: string
+  required?: boolean
+  helper?: string
+  unitError?: string
+  width?: ComponentProps<typeof Stack>['width']
+  placeholderUnit?: string
+  autoFocus?: boolean
+  error?: boolean
+  success?: boolean
+  label?: string
+  labelInformation?: ReactNode
+  step?: number | string
+  readOnly?: boolean
+}
+
+export const UnitInput = ({
+  id,
+  name = '',
+  maxValue = 99999,
+  minValue = 1,
+  autoFocus = false,
+  size = 'large',
+  placeholder = '0',
+  placeholderUnit = 'Select item',
+  onChange,
+  onChangeUnitValue,
+  value,
+  unitValue,
+  selectInputWidth = 200,
+  disabled = false,
+  options,
+  className,
+  label,
+  step = 1,
+  error,
+  notice,
+  required,
+  helper,
+  unitError,
+  success,
+  'data-testid': dataTestId,
+  width,
+  labelInformation,
+  readOnly,
+}: UnitInputProps) => {
+  const [hasFocus, setHasFocus] = useState(false)
+  const [hovered, setHovered] = useState(false)
+  const [val, setVal] = useState(value)
+  const sentiment = useMemo(() => {
+    if (error) {
+      return 'danger'
+    }
+    if (success) {
+      return 'success'
+    }
+
+    return 'neutral'
+  }, [error, success])
+
+  useEffect(() => {
+    const selectBar = document.getElementById('select-bar')
+
+    if (selectBar) {
+      selectBar.style.backgroundColor = 'transparent'
+      selectBar.style.border = 'none'
+      selectBar.style.borderRadius = '0px'
+    }
+  }, [])
+
+  return (
+    <Stack gap={0.5}>
+      {label ? (
+        <Stack direction="row" gap={0.5} alignItems="center">
+          {label ? (
+            <Text
+              as="label"
+              variant={size === 'large' ? 'bodyStrong' : 'bodySmallStrong'}
+              disabled={disabled}
+            >
+              {label}
+            </Text>
+          ) : null}
+          {required && label ? (
+            <Icon name="asterisk" sentiment="danger" size={8} />
+          ) : null}
+          {labelInformation && label ? labelInformation : null}
+        </Stack>
+      ) : null}
+      <UnitInputWrapper
```

Comment:
There is too many borders 

<img width="228" alt="Screenshot 2024-04-30 at 09 58 05" src="https://github.com/scaleway/ultraviolet/assets/15812968/eaf1f974-3f22-4d10-9639-dd221428b8b8">

It should have one unique border + a border separation between both components